### ISSUE TITLE
Missing Instruments when importing from MIDI

### ISSUE THREAD

**Issue by miclol** [OpenNBS/OpenNoteBlockStudio#400]
**Describe the bug**
Instruments seem to be omitted by NBS when imported via MIDI.

**To Reproduce**
1. Extract the Midi File from here: [ttlo.zip](https://github.com/OpenNBS/OpenNoteBlockStudio/files/11549331/ttlo.zip)
2. Import it to the latest version of NBS
3. Check the Tracks and Instruments Tabs before importing the Midi
4. Notice that the Square Synthesizer Instrument is never addressed in the instruments tab but is in the tracks tab even though there are notes in the track that should not be omitted.
5. Even after you import the file the instrument is still missing.

**Expected behavior**
The Square Synthesizer Instrument to be in the imported result.

**Screenshots**
<img width="308" alt="image" src="https://github.com/OpenNBS/OpenNoteBlockStudio/assets/64191125/8c6219ac-7566-4604-a01e-80d2c979f4d7">
<img width="829" alt="image" src="https://github.com/OpenNBS/OpenNoteBlockStudio/assets/64191125/18f99480-7cf1-4cbd-a313-a3e532b01431">

**Comment by Bentroen** [comment]
Thank you for the detailed report!

NBS currently doesn't support multiple tracks mapped to a single channel – the instrument list is always based on the number of channels present in the file, so there's no proper support for stuff like mid-track instrument changes. I believe this is the issue at play here – it correctly identifies the track but it's not imported as a separate instrument.

Do #179 and #194 describe the change you're proposing?

**Comment by miclol** [comment]
Yes, it seems like the problem comes from Musescore creating unused channels in the MIDI files and overriding them with new instruments when it eventually runs out of MIDI channels to use. So even though the instrument count is lower than 16, it replaced the instrument in the same channel anyway.

**Comment by Bentroen** [comment]
Duplicate of #179
